packages/cli/lib/LocalDevManagerV2.js

```diff
@@ -133,12 +131,64 @@ class LocalDevManagerV2 {
     });
   }
 
-  async devServerSetup(componentsByType) {
+  logUploadWarning(reason) {
+    const currentDefaultAccount = getConfigDefaultAccount();
+    const defaultAccountId = getAccountId(currentDefaultAccount);
+
+    logger.log();
+    logger.warn(i18n(`${i18nKey}.uploadWarning.header`, { reason }));
+    logger.log(i18n(`${i18nKey}.uploadWarning.stopDev`));
+    if (this.targetAccountId !== defaultAccountId) {
+      logger.log(
+        i18n(`${i18nKey}.uploadWarning.runUploadWithAccount`, {
+          accountId: this.targetAccountId,
+        })
+      );
+    } else {
+      logger.log(i18n(`${i18nKey}.uploadWarning.runUpload`));
+    }
+    logger.log(i18n(`${i18nKey}.uploadWarning.restartDev`));
+  }
+
+  compareLocalProjectToDeployed(runnableComponents) {
```

Comment:
This compares the contents of `deployedBuild.subbuildStatuses` to see if we have any components or subcomponents locally that don't exist in the deployed build. Currently this is only checking for CRM cards, but we can eventually expand it to check for more than that if we want.